src/ceph-volume/ceph_volume/process.py

```diff
@@ -113,22 +88,10 @@ def run(command, **kw):
         **kw
     )
 
-    if stdin:
-        process.communicate(stdin)
-    while True:
-        reads, _, _ = select(
-            [process.stdout.fileno(), process.stderr.fileno()],
-            [], []
-        )
-        log_descriptors(reads, process, terminal_logging)
-
-        if process.poll() is not None:
-            # ensure we do not have anything pending in stdout or stderr
-            log_descriptors(reads, process, terminal_logging)
-
-            break
-
-    returncode = process.wait()
+    stdout, stderr = process.communicate(stdin)
+    log_output('stdout', stdout, terminal_logging)
+    log_output('stderr', stderr, terminal_logging)
+    returncode = process.poll():
```

Comment:
this change tries to address the failure in https://jenkins.ceph.com/job/ceph-volume-prs-lvm-centos7-bluestore-create/27/consoleFull#-1970939011c19247c4-fcb7-4c61-9a5d-7e2b9731c678

```
Running command: sudo ceph-osd --cluster ceph --osd-objectstore bluestore --mkfs -i 0 --monmap /var/lib/ceph/osd/ceph-0/activate.monmap --keyfile * --bluestore-block-db-path /dev/journals/journal1 --osd-data /var/lib/ceph/osd/ceph-0/ --osd-uuid 055a8ab3-abf4-48f1-9ac6-b095105c7890 --setuser ceph --setgroup ceph


STDERR:

Traceback (most recent call last):
  File "/sbin/ceph-volume", line 6, in <module>
    main.Volume()
  File "/usr/lib/python2.7/site-packages/ceph_volume/main.py", line 37, in __init__
    self.main(self.argv)
  File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 59, in newfunc
    return f(*a, **kw)
  File "/usr/lib/python2.7/site-packages/ceph_volume/main.py", line 144, in main
    terminal.dispatch(self.mapper, subcommand_args)
  File "/usr/lib/python2.7/site-packages/ceph_volume/terminal.py", line 131, in dispatch
    instance.main()
  File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/main.py", line 38, in main
    terminal.dispatch(self.mapper, self.argv)
  File "/usr/lib/python2.7/site-packages/ceph_volume/terminal.py", line 131, in dispatch
    instance.main()
  File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/create.py", line 57, in main
    self.create(args)
  File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 16, in is_root
    return func(*a, **kw)
  File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/create.py", line 21, in create
    Prepare([]).prepare(args)
  File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 16, in is_root
    return func(*a, **kw)
  File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/prepare.py", line 231, in prepare
    fsid=osd_fsid,
  File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/prepare.py", line 72, in prepare_bluestore
    db=db
  File "/usr/lib/python2.7/site-packages/ceph_volume/util/prepare.py", line 224, in osd_mkfs_bluestore
    process.run(command, obfuscate='--keyfile', stdin=keyring)
  File "/usr/lib/python2.7/site-packages/ceph_volume/process.py", line 120, in run
    [process.stdout.fileno(), process.stderr.fileno()],
ValueError: I/O operation on closed file
```

Reply:
it introduces an invalid syntax though:
    File "/usr/lib/python2.7/site-packages/ceph_volume/process.py", line 94
      returncode = process.poll():
                               ^
    SyntaxError: invalid syntax

Reply:
Also, lets try a way to keep these things around, because it is meant to report output as it happens. With the proposed approach we would need to wait until the very end to output anything. 

I believe that the correct thing to do here is to use `process.call` not `process.run`. And to remove the ability to pass `stdin` to `process.run` because as implemented it will never work (once `communicate()` is called the descriptors are closed)

Reply:
@alfredodeza thanks. probably we can use the "stdin" property for writing the stdin of the `Popen` instance?